src/backup.py

```diff
@@ -454,8 +454,12 @@ def _build_backup_command(
     paths = _paths_to_args(backup_paths)
     tar_command = f"tar -c {paths}"
     gpg_command = (
-        f"gpg --batch --no-symkey-cache --passphrase-file '{passphrase_file}' --symmetric"
+        f"gpg --batch --no-symkey-cache "
+        f"--passphrase-file '{passphrase_file}' "
+        f"--symmetric --cipher-algo AES256 "
+        f"--openpgp"
```

Comment:
From the man page I read:
```
Reset all packet, cipher and digest options to strict OpenPGP behavior.  This option implies --allow-old-cipher-algos.  Use this option to reset all previous options like --s2k-*, --cipher-algo, --digest-algo and --compress-algo to OpenPGP compliant values. All PGP workarounds are disabled.
```

So this doesn't seem right to me as it is negating the previous `--cipher-algo` option ?

Reply:
Curious. I ran a test without this parameter, and the output was different: 'GPG symmetrically encrypted data (3DES cipher).' So, I think it is being considered.

Reply:
```
amanda@amoreira:~/Canonical/synapse-operator$ echo "oi" |gpg --batch --no-symkey-cache --passphrase "banana" --symmetric --cipher-algo AES256 --openpgp > test.txt
amanda@amoreira:~/Canonical/synapse-operator$ python
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import magic
>>> magic.from_file("test.txt")
'GPG symmetrically encrypted data (AES256 cipher)'
>>> 
amanda@amoreira:~/Canonical/synapse-operator$ rm -f test.txt
amanda@amoreira:~/Canonical/synapse-operator$ echo "oi" |gpg --batch --no-symkey-cache --passphrase "banana" --symmetric --openpgp > test.txt
amanda@amoreira:~/Canonical/synapse-operator$ python
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import magic
>>> magic.from_file("test.txt")
'GPG symmetrically encrypted data (3DES cipher)'
>>> 
```

Reply:
It is also different if you remove `--openpgp` from the initial command:
```
┌─[2024-11-25 15:55:45 t14 ~]
└─╼ echo "oi" |gpg --batch --no-symkey-cache --passphrase "banana" --symmetric --cipher-algo AES256 > test.txt

┌─[2024-11-25 15:55:50 t14 ~]
└─╼ python3
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import magic
>>> magic.from_file("test.txt")
'PGP symmetric key encrypted data - AES with 256-bit key salted & iterated - SHA512 .'
>>>
```

I would avoid using options without explicitly having something in mind. Have a look at the "INTEROPERABILITY" section of the official website: https://gnupg.org/documentation/manuals/gnupg-2.0/GPG-Examples.html

I'm not sure why we're handling things related to openpgp for a direct symmetric encryption.

Reply:
Thanks for raising this issue. I did some investigation and it seems that the output is different but the same command is still compatible in both gpg versions. Also I tested the restore and it seems fine. So I'm reverted the change and updated the unit test properly.